dexonline/prototype.py

```diff
@@ -112,47 +161,135 @@ def oltenizare(doc):
             phrase_to_return += new_phrase[i]
 
         else:
-            if new_phrase[i-1] in end_of_phrase:
+            if new_phrase[i - 1] in end_of_phrase:
                 phrase_to_return += " " + new_phrase[i].capitalize()
-            elif new_phrase[i-1][-1] == "-":
+            elif new_phrase[i - 1][-1] == "-":
                 phrase_to_return += new_phrase[i]
             else:
                 phrase_to_return += " " + new_phrase[i]
 
     return phrase_to_return
 
 
+def synonyms_builder(token, pos_wanted):
+    """
+    This function will extract each synonym found for a certain word at a
+    base form.
+    Returns:
+        inflection_possibilities = a list of inflection id s for matching
+                                   with the inflection found in input
+        candidates_synonyms_base_form = a list of synonyms at a base form
+    """
+
+    token_text = re.sub('[^a-zA-ZăâîșțĂÂÎȘȚ]', '', token.text.lower())
+    inflected_forms = all_inflected_forms.find_all_inflected_forms(token_text)
+    
+    inflection_possibilities = find_inflection_possibilites(token, inflected_forms, pos_wanted)
+    possible_lexeme_ids = find_lexeme_ids(inflected_forms)
+    lexeme_ids = find_matching_lexemeIds(token, possible_lexeme_ids, pos_wanted)
+    entry_ids = find_entryIds(lexeme_ids)
+    tree_ids = find_treeIds(entry_ids)
+    meaning_ids = find_meaningIds(tree_ids)
+
+    candidate_synonyms_base_form = []
+
+    for meaningId in meaning_ids:
+        possible_synonyms = synonyms.find_synonyms(meaningId)
+        if possible_synonyms != ["no synonyms"]:
+            for synonym in possible_synonyms:
+                syn_to_add = re.sub('[^a-zA-ZăâîșțĂÂÎȘȚ ]', '', synonym[1]).split(" ")
+                
+                for syn in syn_to_add:
+                    syn_to_add_helper = all_inflected_forms.find_all_inflected_forms(
+                                syn,
+                                unidentified={"lexemeId": "UNKNOWN"}
+                            )
+                    if syn_to_add == ["UNKOWN"]:
+                        break
+
+                    syn_tuple = (syn, syn_to_add_helper[0].get("lexemeId", "dummy"))
+                    if syn_tuple not in candidate_synonyms_base_form and syn_tuple[0] != token_text:
+                        candidate_synonyms_base_form.append(syn_tuple)
+
+    candidate_synonyms_base_form = [syn for i, syn in enumerate(candidate_synonyms_base_form) if i == 0 or syn[1] != candidate_synonyms_base_form[i-1][1]]
+
+    return inflection_possibilities, candidate_synonyms_base_form
+
+
+def is_valid_for_syn(token):
+    if token.pos_ == "PUNCT":
+        return False
+    if "aux" in token.dep_:
+        return False
+    if not token.text.isalpha():
+        return False
+    return True
+
+def get_synonyms(token):
+    if is_valid_for_syn(token):
```

Comment:
sugestie:
```
if not is_valid(token):
  return []
....
```
e bine ca o functie sa primeasca un singur tip de date si sa returneze tot un singur tip de date;
in cazul tau acum, tipul de date e acelasi din pct de vedere formal List[str], dar nu e acelasi din pct de vedere semantic; as sugera sa nu returnezi mesaje de eroare aici